### PR TITLE
[15.0][3927][MIG] purchase_post_code_propose_address_sst: Migration to 15.0

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**
[3927](https://www.quartile.co/web#cids=3&menu_id=505&action=1457&model=project.task&view_type=form&id=3927)
Depends on:https://github.com/qrtl/rmm-custom/pull/95

### CODE REVIEW

File: purchase_post_code_propose_address_sst/i18n/ja.po [hunk 1]
```diff
@@ -1,97 +1,102 @@
...
-#: model:ir.ui.view,arch_db:purchase_post_code_propose_address_sst.purchase_order_form
+#: model_terms:ir.ui.view,arch_db:purchase_post_code_propose_address_sst.purchase_order_form
+msgid "<span class=\"oe_read_only\">〒</span>"
+msgstr ""
+
+#. module: purchase_post_code_propose_address_sst
+#: model_terms:ir.ui.view,arch_db:purchase_post_code_propose_address_sst.purchase_order_form
 msgid "Address"
-msgstr "アドレス"
+msgstr ""
```
Comment (yostashiro) [inline purchase_post_code_propose_address_sst/i18n/ja.po]: @AungKoKoLin1997 Can you please revive the translation where the original string hasn't changed.

File: purchase_post_code_propose_address_sst/models/purchase_order.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+                    "message": _("Post code should be 7 digits."),
+                }
+            }
+        return field, msg
+
+    def write(self, vals):
+        res = super(PurchaseOrder, self).write(vals)
+        for order in self:
+            if (
+                not self.is_default_partner(order.partner_id.id)
```
Comment (kanda999) [inline purchase_post_code_propose_address_sst/models/purchase_order.py:101]: is_default_partner depends on purchase_ext_sst module.

**Review by kanda999 — CHANGES_REQUESTED** [review]

File: purchase_post_code_propose_address_sst/models/purchase_order.py (hunk 2 above)
Comment (kanda999) [reply purchase_post_code_propose_address_sst/models/purchase_order.py:101]: Sorry forget about my latest comment

File: purchase_post_code_propose_address_sst/models/purchase_order.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        msg = {}
+        field = jaconv.z2h(zipcode, ascii=True, digit=True).replace("-", "")
+        if not field.isdigit():
+            field = False
+            msg = {
+                "warning": {
+                    "title": _("Error"),
+                    "message": _("Only digits are allowed."),
+                }
+            }
```
Comment (kanda999) [inline purchase_post_code_propose_address_sst/models/purchase_order.py:86]:
```suggestion
            }
            return field, msg
```
Put false in field, so it calls error with `len(field)`.

**Review by kanda999 — CHANGES_REQUESTED** [review]

File: purchase_post_code_propose_address_sst/models/purchase_order.py (hunk 3 above)
Comment (kanda999) [reply purchase_post_code_propose_address_sst/models/purchase_order.py:86]: My suggestion may not be enough.  
We need to refactor this method.

**Review by kanda999 — APPROVED** [review]

File: purchase_post_code_propose_address_sst/models/purchase_order.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    _inherit = "purchase.order"
+
+    country_id = fields.Many2one(
+        "res.country", string="Country", default=lambda self: self.env.ref("base.jp")
+    )
+    state_id = fields.Many2one("res.country.state", string="Prefecture")
+    city = fields.Char()
+    street = fields.Char()
+    street2 = fields.Char()
+    zipcode = fields.Char(string="Post Code (Search)")
```
Comment (yostashiro) [inline purchase_post_code_propose_address_sst/models/purchase_order.py:26]: @kanda999 @AungKoKoLin1997 Can you consider splitting this field and a few methods (`_onchange_zipcode()` and `check_zipcode()`) into a mixin model in a separate module, and then inherit that model in the purchase order class?  The new module with the mixin model can include the extension to res.partner (let this model inherit the mixin) and be put in the Japan localization repo.  What do you think?

Comment (AungKoKoLin1997) [reply purchase_post_code_propose_address_sst/models/purchase_order.py:26]: I think it is a good idea and I will follow up.

**Pull request merged**